Add a blog post for the community introducing the WEP process. (#1116)
writethedocs/www: docs/blog/introducing-weps.rst

Introducing Write the Docs Enhancement Proposals
================================================

> Write the Docs has been growing each year for the past 7,

Reviewer: ```suggestion
Write the Docs has been growing each year since we had the first conference in 2013,
```

and the scope of the community has expanded dramatically.
We started with a single conference,
and now we have multiple conferences,
over 50 meetups around the world,
multiple online spaces,

> and many other ongoing projects.

Reviewer 2: ```suggestion
Write the Docs began seven years ago with a single conference.  
It has now grown to multiple conferences worldwide, over 50 international meetups and many other projects. 
```

Reviewer 2: Edit as needed :-p

Reviewer: this is confusing, where is the comma and where is the period? sometimes the suggestion module looks really weird....

Author: I didn't quite understand this either. I think it was trying to replace the entire paragraph. I've taken that copy and applied it in a commit.

> We have often followed open source in our approach to things,

Reviewer: ```suggestion
We have often followed the open-source philosophy in our community practices,
```

keeping a loose governance model and mostly letting the work speak for itself.
However,

> we've been noticing a number of growing pains with this approach.

Reviewer 2: ```suggestion
we've been experiencing a number of growing pains with this approach.
```

> It makes it much harder for new people to understand and join the community in leadership positions,

Reviewer: what is "it"?

since they don't understand how the community actually works.

> Even folks doing a lot of work in the community are often lost or confused when it comes to other ongoing projects.

Reviewer 3: How about elaborating what we refer to when we refer to "ongoing projects"?

e.g., the job board, salary survey, newsletter, etc ...

Author: I've updated this copy to make it a bit clearer. It was more trying to explain that folks in one part of the organization don't necessarily know what other folks are working on. 👍

All organizations have power structures,

> some are just not documented. [#]_

Reviewer 2: What is this `[#]_`?

Author: Footnote.

Instead of trying to reinvent the wheel,

> we've once again taken an idea from the open source ecosystem:

Reviewer: ```suggestion
we've once again taken an idea from the open-source ecosystem:
```

Author: Pretty sure open source isn't hyphenated when I see it used normally.  This is why we should have a style guide :)

Reviewer: I think that when it's used as an adjective it's hyphenated, but as long as we're consistent *ahem* I'm ok either way...

Write the Docs Enhancement Proposals,

> or WEP's.

Reviewer: ```suggestion
or WEPs.
```

What is a Write the Docs Enhancement Proposal (WEP)
---------------------------------------------------

> WEP's are a formalized process for making major policy changes in the community.

Reviewer: ```suggestion
The WEP system is a formalized process for making major policy changes in the community.
```

> Previously there was fuzzy consensus on a Slack channel or at a conference that defined our governace.

Reviewer 3: ```suggestion
Previously there was fuzzy consensus on a Slack channel or at a conference that defined our governance.
```

(a typo! 😅 )

> Previously there was fuzzy consensus on a Slack channel or at a conference that defined our governance.

Reviewer: ```suggestion
Previously, our governance was usually defined with some kind of fuzzy consensus on a Slack channel or a meeting at a conference.
```

> Going forward, there will be a written record that people have had time to review and comment on.

Reviewer: ```suggestion
Going forward, we want to establish a written record that people have time to review and comment on.
```

Importantly, 
there will also be records of the conversation that lead to a change.

> We hope that WEP's will make it explicit how major changes are made inside the community.

Reviewer: ```suggestion
We hope that WEPs will make the process of making major changes inside the community more explicit and collaborative.
```

> New contributors can propose them with a well-defined process,

Reviewer: ```suggestion
Any contributors can propose WEPs with a well-defined process,
```

> but can also see WEP's proposed by existing contributors.

Reviewer: ```suggestion
and the whole community can follow the process of proposing and implementing WEPs.
```

> WEP's are the first step towards having a more open and defined model for governance,

Reviewer: ```suggestion
WEPs are the first major step towards having a more open and defined model for governance,
```

but certainly not the last.
All further governance changes are planned to go through the WEP process,

> so it is truly foundational for ongoing work on how the community is managed.

Reviewer: ```suggestion
so it is truly foundational for future improvements to our community management processes.
```

What is the WEP process?
------------------------

> The WEP process is defined in `WEP0`_,

Reviewer 2: And this needs defining I think.

the first ever WEP.
The process is currently undergoing review,
following the draft WEP process.

> This is a bit circular,

Reviewer: ```suggestion
This might appear circular at first glance,
```

but we hope that it will showcase how the process works as we adopt it.

The high-level overview of the WEP process is as follows,
from the perspective of someone proposing a WEP:

> * Talk to people about your idea, and get feedback from other community members

Reviewer: change to numbered list?

> * Draft and submit a Pull Request on the WEP GitHub Repository, it is now a **draft**

Reviewer: ```suggestion
* Draft and submit a Pull Request on the WEP GitHub Repository, which submits the WEP as a **draft**
```

> * The **WEP team** then reviews the Draft WEP to make sure it passes a few tests:

Reviewer 2: Maybe something more like 

... to make sure it has the following characteristics

> * The **WEP team** then reviews the Draft WEP to make sure it passes a few tests:

Reviewer: ```suggestion
* The **WEP team** then reviews the Draft WEP to make sure it falls within the WEP guidelines:
```

    * Is the WEP well-formed? Does it have all the required sections?
    * Is the WEP specific enough that the outcome of approving it is clear?

>     * Is the WEP on-topic for the community?

Reviewer: ```suggestion
    * Is the proposed change within the scope of the WEP process?
```

* After the WEP team accepts the draft, it is now **ready-for-discussion**.
* The WEP team then sets a deadline for discussion, currently 30 days, for community feedback and review
* Once the WEP is **ready-for-discussion** and has a deadline, the Core Ops team announce the WEP in a few defined channels, where interested community members can subscribe

>     * The ``WEP-announcements`` Slack channel 

Reviewer: should we also make a 'WEP-discussions' channel, where people can raise ideas? I am assuming that WEP-announcements will be locked for admin posts only

Author: I don't think we can actually lock any channels in Slack other than general :/

Reviewer: in that case let's just have a "WEP" channel, without a suffix, since it'll end up being not-just-announcements

Author: We can try and enforce just having announcements, but we'll see?

    * The ``WEP-announcements`` Email list

> * During this month-long period, as there is feedback, the WEP is updated with any changes, and the community works towards a consensus on a path forward

Reviewer: ```suggestion
* During this month-long period, the WEP is updated based on incoming feedback, and the community works towards a consensus on the proposed change
```

* At the deadline, the Core Ops team determines if consensus has been reached
    * If there is consensus, the WEP is **Accepted**, it will be implemented as specified
    * If there is no consensus, the WEP is either **Rejected** or **Extended** for another month. Extension is available when it seems further activitiy could lead to consensus.
* Once the WEP decision is made, another announcement is sent to the defined announcement channels

> 

Reviewer: Maybe add a sub-heading here "What's next" or "What should we do now?" to get people's attention to the bottom line of "here's WEP0, please review it, learn about it, and ask questions THERE"

If you want to understand the full policy,
we invite you to review and comment on `WEP0`_.
This is a new process,
but we hope it will start to address questions of how change happens in the community. 

>  

Reviewer: Maybe add a "if you have questions about the process, please add them as comments to the WEP0 pull request, or email us at support". Otherwise, the blog post seems to end on a weak note...

Introducing Write the Docs Enhancement Proposals
================================================

Write the Docs began seven years ago with a single conference.  
It has now grown to multiple conferences worldwide,
over 50 international meetups,
and `many other projects`_.

> We have often followed the open-source philosophy in our community practices,

Reviewer 4: ```suggestion
We have often followed the open source philosophy in our community practices,
```
If we're going for non-hyphenated, let's be consistent :)

or WEPs.

What is a Write the Docs Enhancement Proposal (WEP)
---------------------------------------------------

The WEP system is a formalized process for making major policy changes in the community.
Previously, our governance was usually defined with some kind of fuzzy consensus on a Slack channel or a meeting at a conference.
Going forward, we want to establish a written record that people have time to review and comment on.
Importantly, 
there will also be records of the conversation that lead to a change.

We hope that WEPs will make the process of making major changes inside the community more explicit and collaborative.
Any contributors can propose WEPs with a well-defined process,
and the whole community can follow the process of proposing and implementing WEPs.
WEPs are the first major step towards having a more open and defined model for governance,
but certainly not the last.
All further governance changes are planned to go through the WEP process,
so it is truly foundational for future improvements to our community management processes.

What is the WEP process?
------------------------

The WEP process is defined in `WEP0`_,
the first ever WEP.

> The process is currently undergoing review,

Reviewer 4: ```suggestion
WEP0 itself is currently ready for review,
```
